packages/instrumentation-tedious/src/instrumentation.ts

```diff
@@ -207,12 +266,28 @@ export class TediousInstrumentation extends InstrumentationBase<TediousInstrumen
           thisPlugin._diag.error('Expected request.callback to be a function');
         }
 
-        return api.context.with(
-          api.trace.setSpan(api.context.active(), span),
-          originalMethod,
-          this,
-          ...arguments
-        );
+        const runUserRequest = () => {
+          return api.context.with(
+            api.trace.setSpan(api.context.active(), span),
+            originalMethod,
+            this,
+            ...arguments
+          );
+        };
+
+        const cfg = thisPlugin.getConfig();
+        const shouldInject =
+          cfg.enableTraceContextPropagation &&
+          thisPlugin._shouldInjectFor(operation);
+
+        if (!shouldInject) return runUserRequest();
```

Comment:
The previous code and this code path will `return` the return value of the `originalMethod()` call, which is typically what is wanted when wrapping.  The `_injectContextInfo()` code path below this will *not*.  I gather that won't matter for this instrumentation because, at least currently, every patched method's return type is `void`.